### PR TITLE
[smoke-test] Add e2e test for release upgrade workflow

### PR DESCRIPTION
**Pull request by runtian-zhou**
### Description
This PR added an e2e test for release upgrade workflow. To make the workflow work, a few modifications has been made:
1. Fix the overflowing issue in the release generation script by splitting vectors into smaller chunks to avoid Move's size limit for constant.
2. Added an option for the aptos cli to emit the script for the testnet. This is needed because testnet/smoke-test environment has different ways to execute on chain governance proposal.

This tests the following workflow
1. Create a network with aptos framework generated from last committed release (using `testnet.mrb`)
2. Update the gas limit to be able to execute the upgrade script.
3. Generate the release bundle update script via `aptos-cli`. This will generate the release from the current HEAD.
4. Commit the upgrade script using root account
5. Make sure the mint and transfer can still work after the update.

This isn't ideal as we didn't fully test the staking and voting process. Instead we bypassed the voting process by using the root account key for testnet. In the future we might want to be able to test by using staking/voting from `aptos-cli`.

### Test Plan
Added a new test.

- - -
This change is [Reviewable](https://reviewable.io/reviews/aptos-labs/aptos-core/4898)

### CODE REVIEW

**Comment by runtian-zhou** [comment]
cc @davidiw how did you upgrade Aptos Framework on testnet? How did you bypass the max transaction size issue?

File: testsuite/smoke-test/src/upgrade.rs [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+        write!(buf, "{}u8,", b).unwrap();
+    }
+    write!(buf, "]").unwrap();
+    buf
+}
+
+#[tokio::test]
+/// This test verifies the flow of aptos framework upgrade process.
+/// i.e: The network will be alive after applying the new aptos framework release.
+async fn test_upgrade_flow() {
```
Comment (davidiw) [inline testsuite/smoke-test/src/upgrade.rs:36]: this is only a partial upgrade, we have a lot of upgrades to do, see https://github.com/aptos-labs/aptos-genesis-waypoint/tree/main/testnet/proposals/sources/221007-complete-upgrade

we need to do
gas -> stdlib -> aptos stdlib -> aptos framework -> token -> enable any new features

**Review by davidiw — COMMENTED** [review]
great start -- though I do wonder what we're getting in this over the move e2e test.

File: testsuite/smoke-test/src/upgrade.rs [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+        ])
+        .output()
+        .unwrap()
+        .status
+        .success());
+
+    // Two transactions has been emitted to root account.
+    *env.aptos_public_info().root_account().sequence_number_mut() += 2;
+
+    check_create_mint_transfer(&mut env).await;
```
Comment (movekevin) [inline testsuite/smoke-test/src/upgrade.rs:159]: Let's also do module publishing here as well for additional testing post-upgrade

**Review by movekevin — APPROVED** [review]
Amazing work! Thanks a lot for taking a crack at this and also fixing the proposal generation CLI!

File: testsuite/smoke-test/src/upgrade.rs (hunk 1 above)
Comment (runtian-zhou) [reply testsuite/smoke-test/src/upgrade.rs:36]: I'll add the full upgrade flow.

**Comment by runtian-zhou** [comment]
> great start -- though I do wonder what we're getting in this over the move e2e test.

I think this would allow us to test the new aptos framework release package with the new binary. Ideally we need to add new feature we would like to enable in this release cycle and test it towards the end as well.

**Comment by runtian-zhou** [comment]
Interestingly I wasn't able to upgrade the aptos-token module and got an error `CONSTRAINT_NOT_SATISFIED`. Probably sth wrong with the package metadata? cc @wrwg

**Comment by runtian-zhou** [comment]
Is it possible for us to run this test as nightly job rather than land blocking? cc @rustielin

**Review by gregnazario — APPROVED** [review]

**Pull request merged**